### PR TITLE
Remove unnecessary casts to raw pointers

### PR DESCRIPTION
**Pull request by nnmm**
The references automatically coerce to raw pointers.

This also improves the signal-to-noise ratio on the `as` keyword, which is often an
antipattern (e.g. for many numeric casts).

### CODE REVIEW

**Review by esteve — APPROVED** [review]
@nnmm this makes code more readable, thanks!

**Review by esteve — CHANGES_REQUESTED** [review]

File: rclrs/src/lib.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ pub fn spin_once(node: &Node, timeout: Option<Duration>) -> Result<(), RclReturn
 /// This function additionally checks that the context is still valid.
 pub fn spin(node: &Node) -> Result<(), RclReturnCode> {
     // SAFETY: No preconditions for this function.
-    while unsafe { rcl_context_is_valid(&mut *node.context.lock() as *mut _) } {
```
[the remark this answers is not in the document]
Comment (nnmm) [reply rclrs/src/lib.rs:58]: Should be fixed now. It's a Foxy/Galactic difference.

[the remark this answers is not in the document]
Comment (esteve) [reply rclrs/src/lib.rs:58]: > You mean a #[cfg(distro = "foxy")]? We could, but I've got no preference between that or the as _.

Yeah, just to get rid of the cast in the future once Foxy is no longer supported.

[the remark this answers is not in the document]
Comment (nnmm) [reply rclrs/src/lib.rs:58]: Done.

**Review by jhdcs — DISMISSED** [review]
Looks good, now that CI is passing!

File: rclrs/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (esteve) [reply rclrs/src/lib.rs:58]: Thanks.

File: rclrs/build.rs [hunk 2]
```diff
@@ -3,8 +3,23 @@ use std::fs::read_dir;
...
+        panic!(
+            "{} environment variable not set - please source ROS 2 installation first.",
+            env_var
+        );
+    }
+}
 
 fn main() {
+    let distro = get_env_var_or_abort(ROS_DISTRO);
+    println!("cargo:rustc-cfg=distro=\"{distro}\"");
```
Comment (esteve) [inline rclrs/build.rs:21]: Could you rename this to `ros-distro` or `ros_distro`? I know it's nitpicky :sweat_smile:

**Review by esteve — CHANGES_REQUESTED** [review]

File: rclrs/build.rs (hunk 2 above)
Comment (nnmm) [reply rclrs/build.rs]: No problem, done.

**Review by esteve — APPROVED** [review]

File: rclrs/build.rs (hunk 2 above)
Comment (esteve) [reply rclrs/build.rs]: Thanks.

**Pull request merged**